packages/vats/src/core/chain-behaviors.js

```diff
@@ -357,8 +356,8 @@ export const makeChainStorage = async ({
 }) => {
   const bridgeManager = await bridgeManagerP;
   if (!bridgeManager) {
-    console.warn('Cannot support chainStorage without an actual chain.');
-    chainStorageP.resolve(null);
+    console.warn('no bridge so chainStorage will not write.');
+    chainStorageP.resolve(makeMockChainStorageRoot());
```

Comment:
Yay! The `undefined` case for chainStorage was only for the sim chain, I'm pretty sure.

`console.warn` is exactly the way I think this should be treated.

A big part of me would like to tell the type system that this thing is always there so we don't have to keep supporting the sim chain in new code. That might be more involved, though.
cc @michaelfig @0xpatrickdev

Reply:
What would it take to remove sim-chain concerns from chain-behaviors entirely? I figure we could provide a fake `bridge` device here: https://github.com/Agoric/agoric-sdk/blob/af9e1c1d7de068b1e634816da61e9caa2f39948e/packages/vats/src/core/chain-behaviors.js#L319-L328

But I don't know what fidelity it needs or if anything downstream relies on detecting its absence.

Reply:
> What would it take to remove sim-chain concerns from chain-behaviors entirely?

I can't think of any technical obstacles. @michaelfig , maybe you know?

Seems like product (@mitdralla, @sufyaankhan ) should be involved. @sufyaankhan led the charge to hide the sim chain from docs.agoric.com .

Reply:
> > What would it take to remove sim-chain concerns from chain-behaviors entirely?
> 
> I can't think of any technical obstacles. @michaelfig , maybe you know?

No technical obstacles.  Earlier in our history, it was considered onerous to force dapp-developers to install Docker or compile agd themselves.  Nowadays, we rely on Docker without batting an eye.  The times have changed, and sim-chain is a relic we can safely jettison.